common/Utils.hpp

```diff
@@ -13,7 +13,18 @@
 #include "Constants.hpp"
 #include "time.hpp"
 #include <math.h>
+const static bool THROW_DEBUG_EXCEPTIONS = true;
+static inline void debugThrow(std::exception exception) {
```

Comment:
If you wanted, you could have this function take a template argument so it would work for every type of exception.
~~~{.cpp}
template<typename T>
static inline void debugThrow(const T &exception)...
~~~